### PR TITLE
[gps] Update ublox driver and RTCM support (WIP)

### PR DESCRIPTION
**Pull request by fvantienen**
This is an upgrade for the UBlox driver and also some cleanup of the RTCM support.

### CODE REVIEW

**Comment by OpenUAS** [comment]
Chris is our new PPRZ RTCM expert, Will discus this open PR with him, so it can be finalized correctly.

**Comment by OpenUAS** [comment]
Chris is hard at work to get UBX moving base working...

**Comment by OpenUAS** [comment]
All seems to work as it should. tested with uBlox M8P rover and base thanks to Chris in the cold field testflying

File: sw/airborne/subsystems/gps/gps_ubx.h [hunk 1]
```diff
@@ -35,9 +35,9 @@
 
 #ifdef GPS_I2C
 #include "modules/gps/gps_ubx_i2c.h"
-#endif
-
+#else
 #include "mcu_periph/uart.h"
+#endif
```
Comment (OpenUAS) [inline sw/airborne/subsystems/gps/gps_ubx.h]: Not really IMHO... In an airframe we have on GPS on I2C the secondary on UART

File: sw/ground_segment/misc/rtcm2ivy.c [hunk 2]
```diff
@@ -118,16 +111,34 @@ static void ivy_send_message(uint8_t packet_id, uint8_t len, uint8_t msg[])
...
     offset += (packet_size-6);
+    usleep(50000);
 
     if (logger == TRUE) {
       pFile = fopen("./RTCM3_log.txt", "a");
       fprintf(pFile, "%s\n", gps_packet);
       fclose(pFile);
     }
-    printf_debug("Ivy send: %s\n", gps_packet);
+    printf("Ivy send: %s\n", gps_packet);
```
Comment (OpenUAS) [inline sw/ground_segment/misc/rtcm2ivy.c]: Why not only with Debug...

File: sw/ground_segment/misc/rtcm2ivy.c [hunk 3]
```diff
@@ -308,28 +418,24 @@ int main(int argc, char **argv)
...
   printf_debug("Setup RTCM3 callbacks...\n");
   msg_state_init(&msg_state);
+  rtcm3_register_callback(&msg_state, RTCM3_MSG_4072, &rtcm3_4072_callback, &rtcm3_4072_node);
   rtcm3_register_callback(&msg_state, RTCM3_MSG_1005, &rtcm3_1005_callback, &rtcm3_1005_node);
   rtcm3_register_callback(&msg_state, RTCM3_MSG_1077, &rtcm3_1077_callback, &rtcm3_1077_node);
-  rtcm3_register_callback(&msg_state, RTCM3_MSG_1087, &rtcm3_1087_callback, &rtcm3_1087_node);
-
-  rtcm3_register_callback(&msg_state, UBX_NAV_SVIN, &ubx_navsvin_callback, &ubx_nav_svin_node);
-
+  //rtcm3_register_callback(&msg_state, RTCM3_MSG_1087, &rtcm3_1087_callback, &rtcm3_1087_node);
```
Comment (OpenUAS) [inline sw/ground_segment/misc/rtcm2ivy.c]: Why leave 3 commented lines in there?

File: sw/ground_segment/misc/rtcm2ivy.c [hunk 4]
```diff
@@ -308,28 +418,24 @@ int main(int argc, char **argv)
...
+  //rtcm3_register_callback(&msg_state, RTCM3_MSG_1097, &rtcm3_1097_callback, &rtcm3_1097_node);
+  //rtcm3_register_callback(&msg_state, RTCM3_MSG_1127, &rtcm3_1127_callback, &rtcm3_1127_node);
+  rtcm3_register_callback(&msg_state, RTCM3_MSG_1230, &rtcm3_1230_callback, &rtcm3_1230_node);
 
-  // Add IO watch for tty connection
```
Comment (OpenUAS) [inline sw/ground_segment/misc/rtcm2ivy.c]: Why no GLib channel system  option anymore?

File: sw/airborne/subsystems/gps/gps_ubx.c [hunk 5]
```diff
@@ -25,16 +25,20 @@
 #include "led.h"
 
 #ifndef USE_GPS_UBX_RTCM
-#define USE_GPS_UBX_RTCM 0
```
Comment (OpenUAS) [inline sw/airborne/subsystems/gps/gps_ubx.c]: is it not that if USE_GPS_UBX_RTCM only then it should be enabled?

**Comment by OpenUAS** [comment]
@fvantienen now that the moving base small changes are in master, maybe a good moment to finalize the pull request?

**Comment by fvantienen** [comment]
Closing this because of another branch

**Pull request closed without merging**